stan/math/prim/meta/require_generics.hpp

```diff
@@ -473,24 +473,22 @@ using require_any_not_fvar_t
     = require_any_not_t<is_fvar<std::decay_t<Types>>...>;
 
 template <typename T>
-using require_var_or_fvar_t = require_t<is_var_or_fvar<T>>;
+using require_autodiff_t = require_t<is_autodiff<T>>;
 
 template <typename T>
-using require_not_var_or_fvar_t = require_not_t<is_var_or_fvar<T>>;
+using require_not_autodiff_t = require_not_t<is_autodiff<T>>;
```

Comment:
> removing the `require_` prefix because that's implying a use, not a behavior

Can you clarify? I'm not sure i understand how the require part of the name is implying a use over a behaviour. The intent with the `require_` in the name is because all those `require_*`s are trying to emulate a legacy version of C++20 concepts. It's for if you want a universal reference in the function signature for all the weird value types and const combinations. In terms of use/behavior I think my answer is that I think about C++ concepts like compile time contracts (which feels like a behaviour, though idk what that word means in context here)

Reply:
> replace _t with _v because it denotes a boolean value, not a type; this follows the behavior of is_arithmetic_v; enable_if_t ends in _t because it denotes a type (void by default), not a value.
> For example, that'd be not_autodiff_v<T> for true if T is not an autodiff type, and usage would be, for example enable_if_t<not_autodiff_v<T>, U>.

The `require_*_t` metaprogramming stuff is essentially a convoluted / compact way to write `std::enable_if_t<...>` i.e. they also return `void`.

You can think of `require_autodiff_t` as an alias for `enable_if_t<not_autodiff_v<T>, U>`.

Also idt we can use C++ compile time constexpr objects with c++1y (anything _v like how std does) since the windows compiler we require compatibility with for Rtools has a bug for those

Reply:
I misunderstood and thought this was a boolean.  Now that I see what it's doing, I still think it needs to be renamed.  How about `enable_if_autodiff_t` to enable if it's an autodiff type?  That follows the standard library naming conventions more closely than `require_autodiff_t`.  

We can then use `disable_if_autodiff_t` instead of `enable_if_not_autodiff_t`.  

We can build similar ones for primitives, `enable_if_arithmetic_t` and `disable_if_arithmetic_t`.